sdk/src/wallet/migration/chrysalis.rs

```diff
@@ -0,0 +1,310 @@
+                                    log::warn!(
+                                        "Skip migrating public address because of gap: {bech32_address}, index {key_index}"
+                                    );
+                                    continue;
+                                }
+                                highest_public_address_index += 1;
+                            }
+                            account_addresses.push(AccountAddress {
+                                address: bech32_address,
+                                key_index,
+                                internal,
+                                used: !address["outputs"].as_object().unwrap().is_empty(),
+                            })
+                        }
+                    }
+                    let (internal, public): (Vec<AccountAddress>, Vec<AccountAddress>) =
+                        account_addresses.into_iter().partition(|a| a.internal);
+
+                    new_accounts.push(AccountDetailsDto {
+                        index: account_data["index"].as_u64().unwrap() as u32,
+                        coin_type: IOTA_COIN_TYPE,
+                        alias: account_data["alias"].as_str().unwrap().to_string(),
+                        public_addresses: public,
+                        internal_addresses: internal,
+                        addresses_with_unspent_outputs: Vec::new(),
+                        outputs: HashMap::new(),
+                        unspent_outputs: HashMap::new(),
+                        transactions: HashMap::new(),
+                        pending_transactions: HashSet::new(),
+                        locked_outputs: HashSet::new(),
+                        incoming_transactions: HashMap::new(),
+                        native_token_foundries: HashMap::new(),
+                    })
+                }
+            }
+        }
+    }
+    // Accounts must be ordered by index
+    new_accounts.sort_unstable_by_key(|a| a.index);
+    Ok((new_accounts, secret_manager_dto))
+}
+
+fn get_chrysalis_data(chrysalis_storage_path: &Path, password: Option<Password>) -> Result<HashMap<Vec<u8>, String>> {
+    let encryption_key = password.map(storage_password_to_encryption_key);
+    let chrysalis_db = DB::open_default(chrysalis_storage_path)?;
+    let mut chrysalis_data = HashMap::new();
+    // iterate over all rocksdb keys
+    for item in chrysalis_db.iterator(IteratorMode::Start) {
+        let (key, value) = item?;
+
+        let key_utf8 = String::from_utf8(key.to_vec()).map_err(|_| Error::Migration("invalid utf8".into()))?;
+        let value = if let Some(encryption_key) = &encryption_key {
+            let value_utf8 = String::from_utf8(value.to_vec()).map_err(|_| Error::Migration("invalid utf8".into()))?;
+            // "iota-wallet-key-checksum_value" is never an encrypted value
+            if serde_json::from_str::<Vec<u8>>(&value_utf8).is_ok() && key_utf8 != "iota-wallet-key-checksum_value" {
```

Comment:
nit but could maybe rework it with an if let so we don't have to parse it L276

Reply:
Done